### DISCUSSION TITLE
YouTube watch session expired error when using a VPN

### DISCUSSION THREAD

**Discussion by kgarrett67** [FreeTubeApp/FreeTube#6104] [General]
I just download and started trying to use FreeTube on my Linux OpenSUSE Leap 15.5 system. At first I was getting many errors, but solved all but one by setting preferred API Backend to Invidious API. However, one I could not solve is [VIDEO_ERROR] YouTube watch session expired. Please reopen this video. The only way I can watch a video is turn off my VPN. Also I found that I could change back to Local API and watch videos with the VPN turned off.

- Question: Do I have to turn off my VPN in order to watch videos using FreeTube?

I'm using 0.22.0 Beta downloaded and installed from flatpak.

![watch-session-expired-vpn-on](https://github.com/user-attachments/assets/b7f1edd2-537f-4a36-b1b2-1073fe1ca93a)

**Comment by absidue** [comment]
That likely means you are using a VPN IP address that is blocked by YouTube which would also explain your problems with the local API. You can either try using a different VPN IP/server or turn on `Proxy Videos Through Invidious` in the `Player` settings section.

**Reply by kgarrett67** [reply]
Thank you for your help. However, I turned on `Proxy Videos Through Invidious` and the results are the same. I even closed and reopened FreeTube and still receive the error message. I'll try another VPN server, but can't do that right away.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
@kgarrett67 are you still experiencing this?

**Reply by kgarrett67** [reply]
Thank you, @efb4f5ff-1298-471a-8973-3d47447115dc for the follow up.

TBH, I had given up on FreeTube for the moment and discovered that I could use the Tor Browser to watch YouTube videos without logging in on a hit and miss. However, since your reply, I am on another VPN that is closer to where I live and in conjunction turning on `Proxy Videos Through Invidious` I was able to watch 2 YouTube videos so far without the error message. I'll switch back to FreeTube for a few days and give you some feedback.

Incidentally, had there been an update that may have affected the error message?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> Incidentally, had there been an update that may have affected the error message?

Not that i can remember from the top of my head.

I hope this was just an issue with the VPN Server you were using at the time.

**Reply by kgarrett67** [reply]
That's what I was thinking, but at the time, even after changing VPN servers it was acting buggy. But working now...yay!!!

Thank you again for the follow up reply

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
You're welcome, im happy thats it working for you :)

I will close this thread but do no hesitate to comment on this thread when the issue resurfaces

**Comment by dadadies** [comment]
Just downloaded Freetube and getting this error. I tried switching to Invidious API but still getting the same error:

[VIDEO_ERROR] YouTube watch session expired. Please reopen this video.

I cant turn off my VPN as that is what gives me internet. Everything else seems to work except playing the video.

**Reply by kgarrett67** [reply]
I've about given up on FreeTube. YouTube is just making it too difficult to watch without signing in. Since my last post here, FreeTube worked flawlessly for about 2 days, but now it pops up all kinds of errors. The best I've been able to do is use Tor browser with a VPN, but even it's a hit and miss. Sometimes I can watch YouTube videos without signing in and other times it says I need to sign in.

**Reply by absidue** [reply]
Often times the sign in message just means that the IP is blocked because someone mass-downloaded videos on it (the next big FreeTube update will show a better error message explaining that instead of just showing YouTube's message), the reason is that YouTube will let you watch videos while logged in even on a blocked IP address, as they assume that if you are logged in with an account you are a human. That's the disadvantage of using an IP address that is shared with other people, if someone else gets that VPN IP blocked, everyone using that VPN IP receives the block errors.

**Reply by kgarrett67** [reply]
Here is another error I'm getting, among others

`Invidious API Error: SyntaxError: Unexpected token '<',"<html><h"...is  not valid JSON`

**Reply by absidue** [reply]
The Invidious server returned HTML instead of the JSON that FreeTube expected, likely means that instance is temporarily experiencing problems e.g. getting IP blocked by YouTube because too many people are using that instance, so causing too much traffic.
It's not a FreeTube problem.
